### PR TITLE
Delaunay cech utility

### PR DESCRIPTION
**Pull request by VincentRouvreau**

### CODE REVIEW

File: src/Cech_complex/utilities/delaunay_cech_persistence.cpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  Simplex_tree stree;
+
+  Points_off_reader off_reader(off_file_points);
+  std::vector<Point> point_cloud = off_reader.get_point_cloud();
+  Delaunay_complex delaunay_complex_from_file(point_cloud);
+  delaunay_complex_from_file.create_complex(stree, std::numeric_limits< Filtration_value >::infinity(),
+                                            // exact can be false (or true), as default_filtration_value is set to true
+                                            false, true);
+  Gudhi::cech_complex::assign_MEB_filtration(Kernel(), stree, point_cloud, exact_version);
+  stree.prune_above_filtration(max_radius);
```
Comment (mglisse) [inline src/Cech_complex/utilities/delaunay_cech_persistence.cpp:52]: Unlike the alpha-complex, for Delaunay-Cech, we could do a bit better (don't compute filtration for simplices whose faces already have a too large filtration), but that's not important for now.

File: src/Cech_complex/utilities/delaunay_cech_persistence.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  using Delaunay_complex = Gudhi::alpha_complex::Alpha_complex<Kernel>;
+
+  Simplex_tree stree;
+
+  Points_off_reader off_reader(off_file_points);
+  std::vector<Point> point_cloud = off_reader.get_point_cloud();
+  Delaunay_complex delaunay_complex_from_file(point_cloud);
+  delaunay_complex_from_file.create_complex(stree, std::numeric_limits< Filtration_value >::infinity(),
+                                            // exact can be false (or true), as default_filtration_value is set to true
+                                            false, true);
```
Comment (mglisse) [inline src/Cech_complex/utilities/delaunay_cech_persistence.cpp:50]: Computing a Delaunay triangulation looks a bit strange (alpha-complex with a bunch of meaningless arguments), but that's how it is for now, ok.

File: src/Cech_complex/utilities/delaunay_cech_persistence.cpp [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  po::options_description hidden("Hidden options");
+  hidden.add_options()("input-file", po::value<std::string>(&off_file_points),
+                       "Name of an OFF file containing a point set.\n");
+
+  po::options_description visible("Allowed options", 100);
+  visible.add_options()("help,h", "produce help message")(
+      "exact,e", po::bool_switch(&exact),
+      "To activate exact version of Delaunay-Cech complex (default is false, not available if fast is set)")(
+      "fast,f", po::bool_switch(&fast),
+      "To activate fast version of Delaunay-Cech complex (default is false, not available if exact is set)")(
```
Comment (mglisse) [inline src/Cech_complex/utilities/delaunay_cech_persistence.cpp:123]: (I realize that this is probably copied from another utility)
It isn't super clear what exact/fast mean precisely. We should say somewhere:
1. fast: right combinatorics, values can be arbitrarily bad
2. default, safe: values can have a relative error at most XXX
3. exact: true values rounded to double.

File: src/Cech_complex/utilities/delaunay_cech_persistence.cpp [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+using Field_Zp = Gudhi::persistent_cohomology::Field_Zp;
+using Persistent_cohomology = Gudhi::persistent_cohomology::Persistent_cohomology<Simplex_tree, Field_Zp>;
+
+void program_options(int argc, char* argv[], std::string& off_file_points, bool& exact, bool& fast,
+                     std::string& pers_diag_file, Filtration_value& max_radius, int& p,
+                     Filtration_value& min_persistence);
+
+template<class Kernel>
+Simplex_tree create_simplex_tree(const std::string &off_file_points, bool exact_version, Filtration_value max_radius) {
+  using Point = typename Kernel::Point_d;
```
Comment (mglisse) [inline src/Cech_complex/utilities/delaunay_cech_persistence.cpp:39]: In principle, we should be able to construct the Delaunay triangulation with Epick_d always, and only vary the kernel for assign_MEB_filtration. That probably requires converting point_cloud and I don't know how much there is to win here.

**Review by mglisse — APPROVED** [review]

File: src/Cech_complex/utilities/delaunay_cech_persistence.cpp (hunk 3 above)
Comment (VincentRouvreau) [reply src/Cech_complex/utilities/delaunay_cech_persistence.cpp:123]: To be discussed on #1029

**Pull request merged**